### ISSUE TITLE
The behavior of log

### ISSUE THREAD

**Issue by visionxyz** [OpenRLHF/OpenRLHF#429]
I am using the same training setting for the current version and the old version of the code. But I find that the logic for the log is different. 
https://github.com/OpenRLHF/OpenRLHF/blob/cafd60691726ea7dfb474c548347e4a204877026/openrlhf/trainer/dpo_trainer.py#L188
Here, global_step = step // self.strategy.accumulated_gradient.
https://github.com/OpenRLHF/OpenRLHF/blob/cafd60691726ea7dfb474c548347e4a204877026/openrlhf/trainer/dpo_trainer.py#L206
Here, we need, global_step % self.strategy.accumulated_gradient == 0.

But global_step is step // self.strategy.accumulated_gradient, if step is 256 and self.strategy.accumulated_gradient is 256, then global_step is 1. The step needs to be 256*256, then the global_step % self.strategy.accumulated_gradient == 0, which is much longer than we expected.

**Comment by hijkzzz** [comment]
global_step is the model weights update times
we use global_step  to save checkpoint (optimizer/model/data states)

**Comment by visionxyz** [comment]
Thank you so much! I can understand this part since we need to accumulate the gradient. 
https://github.com/OpenRLHF/OpenRLHF/blob/cafd60691726ea7dfb474c548347e4a204877026/openrlhf/trainer/dpo_trainer.py#L201
https://github.com/OpenRLHF/OpenRLHF/blob/cafd60691726ea7dfb474c548347e4a204877026/openrlhf/trainer/dpo_trainer.py#L206
But, if global_step % args.logging_steps == 0, why we need to check global_step % self.strategy.accumulated_gradient == 0?
It would be appreciated, if you could answer my question. 

For example, world size is 8, batch size is 256, mini batch is 1, then the self.strategy.accumulated_gradient is 32, right? If my logging_step is 1, I hope to update the log for every global step. But even if global_step (1) % args.logging_steps (1) == 0, global_step (1) % self.strategy.accumulated_gradient (32) != 0. 
I can not understand this part, I've checked the old code, this part is the same. But how to compute the global_step is different. Old code works fine for wandb, the newest one only uploads limited data.

**Comment by visionxyz** [comment]
I think the current global_step is correct, but we may need to delete code like Line 196 for trainers. 
https://github.com/OpenRLHF/OpenRLHF/blob/6ce02e7079abe77a67c6efb0db53d31e18fff858/openrlhf/trainer/dpo_trainer.py#L196

This is for the old global_step, which looks like the step for the current code.

**Comment by hijkzzz** [comment]
> I think the current global_step is correct, but we may need to delete code like Line 196 for trainers.
> 
> https://github.com/OpenRLHF/OpenRLHF/blob/6ce02e7079abe77a67c6efb0db53d31e18fff858/openrlhf/trainer/dpo_trainer.py#L196
> 
> This is for the old global_step, which looks like the step for the current code.

I have deleted this line in the latest version: https://github.com/OpenRLHF/OpenRLHF/blob/cafd60691726ea7dfb474c548347e4a204877026/openrlhf/trainer/dpo_trainer.py#L188

**Comment by hijkzzz** [comment]
谢谢，您说的是对的。
